plugin/keyring_vault/vault_io.h

```diff
@@ -0,0 +1,54 @@
+#ifndef MYSQL_VAULT_IO_H
+#define MYSQL_VAULT_IO_H
+
+#include <my_global.h>
+#include <logger.h>
+#include "i_vault_io.h"
+#include "vault_parser.h"
+#include "vault_curl.h"
+#include "vault_key_serializer.h"
+
+namespace keyring {
+
+class Vault_io : public IVault_io
+{
+public:
+  Vault_io(ILogger *logger, IVault_curl *vault_curl,
```

Comment:
In my opinion, the statement in Google Coding Standard about not using non-const references at all is very questionable. A few comments on this:
1. The most important one. With pointer you can pass a null_ptr to a function which never expect such value. Moreover, such functions very often do not check for input parameters to be non-null. Why add extra check if there is already an existing mechanism - references.
2. Google's primary argument about using pointers for output parameters is that `func(&value)` is more readable than `func(value)`.  Having address operator before a variable 'should' indicate that it will be used for output. In my opinion, the benefits of this cannot be even compared with drawbacks of the previous parameters. Moreover, there are other ways to indicate that parameter is an output one:
  a. From the function name itself. For instance, `std::getline(is, str)`. The name `get line` suggests 
  that this function must return a line somewhere and here `str` is the only place.
  b. If you really want to improve readability, just use some coding convention for output variable 
  names, e.g. `std::getline(is, out_str)`
3. Very often non-const references indicate that it is not an output parameter but an input one on which the function is allowed to use non-const interface. The same example with `std::getline(is, str)`, `is` is an input parameter passed by non-const reference.
4. In my opinion, the best coding standards is what we already have in C++ core library and STL. Boost is another good example. Pointers instead of references would be very smelly code there.